### PR TITLE
Added retry logic to the cp() method

### PR DESCRIPTION
**Pull request by egor-bryzgalov**
Added retry logic to the cp() method to avoid intermittent build fails.

Related work item: [#757](https://github.com/microsoft/build-task-team/issues/757)

Test examples:
[Failed build with retryCount set to 2.](https://dev.azure.com/v-egbryz/TestProject/_build/results?buildId=1614&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=1f0a8695-ceaa-5eb2-5ab8-df326b6acd5f&l=19)
[Failed build with retryCount set to 2 (debug)](https://dev.azure.com/v-egbryz/TestProject/_build/results?buildId=1615&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=1f0a8695-ceaa-5eb2-5ab8-df326b6acd5f)
[build with retryCount set to 2 (debug, continue on error allowed)](https://dev.azure.com/v-egbryz/TestProject/_build/results?buildId=1613&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=1f0a8695-ceaa-5eb2-5ab8-df326b6acd5f)

### CODE REVIEW

**Comment by DaniilShmelev** [comment]
Could you please explain why this change is needed?

**Comment by egor-bryzgalov** [comment]
@DaniilShmelev This change is intended to avoid build fails in cases when an intermittent issue appears while copying (e.g. connection errors). The default behavior is not changed except for a warning thrown if an error occurred and `continueOnError` is allowed.

**Comment by DaniilShmelev** [comment]
@egor-bryzgalov maybe in this case it makes sense to add some sort of a delay between retries?

File: node/task.ts [hunk 1]
```diff
@@ -744,16 +744,30 @@ export function ls(options: string, paths: string[]): string[] {
  * @param     dest       destination path
  * @param     options    string -r, -f or -rf for recursive and force 
  * @param     continueOnError optional. whether to continue on error
+ * @param     retryCount optional. Retry count to copy the file
  */
-export function cp(source: string, dest: string, options?: string, continueOnError?: boolean): void {
+export function cp(source: string, dest: string, options?: string, continueOnError?: boolean, retryCount: number = 0): void {
```
Comment (anatolybolshakov) [inline node/task.ts:749]: Could you please update docs as well? https://github.com/Microsoft/azure-pipelines-task-lib/blob/master/node/docs/azure-pipelines-task-lib.md#taskcp

**Review by anatolybolshakov — APPROVED** [review]

File: node/task.ts (hunk 1 above)
Comment (egor-bryzgalov) [reply node/task.ts:749]: Thank you! Fixed it.

**Comment by egor-bryzgalov** [comment]
Updated example links

**Review by EzzhevNikita — APPROVED** [review]

**Review by sdobrodeev — APPROVED** [review]

**Review by damccorm — APPROVED** [review]

**Pull request merged**